src/components/Firestore/DocumentListItem.tsx

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  ListItem,
+  ListItemPrimaryText,
+  ListItemSecondaryText,
+  ListItemText,
+} from '@rmwc/list';
+import { firestore } from 'firebase';
+import get from 'lodash.get';
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+import { useFirestoreDocData } from 'reactfire';
+
+const DocumentListItem: React.FC<{
+  reference: firestore.DocumentReference;
+  url: string;
+  queryField?: string;
+}> = ({ queryField, reference, url }) => {
+  const data = useFirestoreDocData<{}>(reference);
```

Comment:
This will open a listener for every document in the list, which is unnecessary. Remember that we already have the document snapshots in Collection.tsx, and you can just pass in `<DocumentListItem queryFieldValue={get(doc.data(), queryField)}>`.

Reply:
@yuchenshi Thank you for the comment! I fixed it.